src/main.c

```diff
@@ -272,7 +273,23 @@ main_loop(void)
 	/* TODO: set listening socket */
 	debug(LOG_NOTICE, "Created web server on %s", config->gw_address);
 
+	if (config->preauth) {
+		debug(LOG_NOTICE, "Preauth is Enabled - Overiding FAS configuration.\n");
```

Comment:
Why do we need to set FAS variables when it is disabled? This feels wrong.

Reply:
PreAuth is a FAS that uses MHD to serve html, so FAS is enabled with very specific settings. So instead of requiring the user to set them we do it for them here making it simple.

Reply:
PreAuth is now enabled with the single line in config:
`	option preauth '/usr/lib/nodogsplash/login.sh'`

This is now safe to enable regardless of any previously set FAS config options. Before this change it was quite complicated to understand and very easy for the user to mess up.